### ISSUE TITLE
Valid TXT records may contain binary data

### ISSUE THREAD

**Issue by dimbleby** [tailhook/dns-parser#22]
The parser insists that the content of a valid TXT record be UTF8, but it ain't necessarily so.

* RFC 1035 talks about "character-string"s without, so far as I can see, ever spelling out what a "character" actually is
  * (I'd have guessed that the intention was printable ASCII)
* RFC6763 (DNS service discovery, which makes use of TXT records) is explicit, though.  See section 6.5:

> The value is opaque binary data.  Often the value for a particular attribute will be US-ASCII [RFC20] or UTF-8 [RFC3629] text, but it is legal for a value to be any binary data.

Here's an example found in the wild:

```
$ dig @8.8.8.8 like.com.sa TXT

; <<>> DiG 9.9.4-RedHat-9.9.4-51.el7 <<>> @8.8.8.8 like.com.sa TXT
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 15148
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;like.com.sa.                   IN      TXT

;; ANSWER SECTION:
like.com.sa.            14399   IN      TXT     "v=spf1 ip4:70.38.11.53 +a +mx +ip4:\184k\180c ?all"

;; Query time: 300 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Mon Oct 23 08:35:50 BST 2017
;; MSG SIZE  rcvd: 97
```

**Comment by tailhook** [comment]
Well, this is a breaking change, but let's go for it. The earlier we fix it the better. Would you like to make a PR?

**Comment by dimbleby** [comment]
This is something-I-noticed rather than something-I-need.  So if you - or anyone else! - would like this sorted sooner rather than later, then please don't wait on me.

But if you're content for this to happen whenever the mood takes me, which probably it eventually will, then sure.

**Comment by tailhook** [comment]
Fixed in #31
